src/chttpd/src/chttpd_httpd_handlers.erl

```diff
@@ -48,27 +48,439 @@ design_handler(<<"_info">>)    -> fun chttpd_db:handle_design_info_req/3;
+                'doc.id' => DocId
+            }};
+        _ ->
+            no_match
+    end;
+
+handler_info('GET', [<<"_session">>], _) ->
     {'session.read', #{}};
-handler_info(#httpd{path_parts=[<<"_session">>], method='POST'}) ->
-    {'session.write', #{}};
-handler_info(#httpd{path_parts=[<<"_session">>], method='DELETE'}) ->
+
+handler_info('POST', [<<"_session">>], _) ->
+    {'session.create', #{}};
+
+handler_info('DELETE', [<<"_session">>], _) ->
     {'session.delete', #{}};
 
-handler_info(#httpd{path_parts=[_Db], method=Method})
-        when Method =:= 'HEAD' orelse Method =:= 'GET' ->
-    {'database-info.read', #{}};
-handler_info(#httpd{path_parts=[_Db], method='POST'}) ->
-    {'document.write', #{}};
-handler_info(#httpd{path_parts=[_Db], method='PUT'}) ->
-    {'database.create', #{}};
-handler_info(#httpd{path_parts=[_Db], method='DELETE'}) ->
-    {'database.delete', #{}};
-
-handler_info(_) -> no_match.
+handler_info('GET', [<<"_up">>], _) ->
+    {'health.read', #{}};
+
+handler_info('GET', [<<"_utils">> | Path], _) ->
+    {'utils.read', #{'file.path' => filename:join(Path)}};
+
+handler_info('GET', [<<"_uuids">>], _) ->
+    {'uuids.read', #{}};
+
+handler_info('GET', [<<"favicon.ico">>], _) ->
+    {'favicon.ico.read', #{}};
+
+
+handler_info(Method, [<<"_", _/binary>> = Part| Rest], Req) ->
+    % Maybe bail here so that we don't trample over a
+    % different url_handler plugin. However, we continue
+    % on for known system databases.
+    DbName = case Part of
+        <<"_dbs">> -> '_dbs';
+        <<"_global_changes">> -> '_global_changes';
+        <<"_metadata">> -> '_metadata';
+        <<"_nodes">> -> '_nodes';
+        <<"_replicator">> -> '_replicator';
+        <<"_users">> -> '_users';
+        _ -> no_match
+    end,
+    if DbName == no_match -> no_match; true ->
+        handler_info(Method, [DbName | Rest], Req)
+    end;
+
+handler_info('GET', [Db], _) ->
+    {'db.info.read', #{'db.name' => Db}};
+
+handler_info('PUT', [Db], _) ->
+    {'db.create', #{'db.name' => Db}};
+
+handler_info('POST', [Db], _) ->
+    {'db.doc.write', #{'db.name' => Db}};
+
+handler_info('DELETE', [Db], _) ->
+    {'db.delete', #{'db.name' => Db}};
+
+handler_info(M, [Db, <<"_all_docs">>], _) when M == 'GET'; M == 'POST' ->
+    {'db.all_docs.read', #{'db.name' => Db}};
+
+handler_info('POST', [Db, <<"_all_docs">>, <<"queries">>], _) ->
+    {'db.all_docs.read', #{'db.name' => Db, multi => true}};
+
+handler_info('POST', [Db, <<"_bulk_docs">>], _) ->
+    {'db.docs.write', #{'db.name' => Db, bulk => true}};
+
+handler_info('POST', [Db, <<"_bulk_get">>], _) ->
+    {'db.docs.read', #{'db.name' => Db, bulk => true}};
+
+handler_info('GET', [Db, <<"_changes">>], _) ->
+    {'db.changes.read', #{'db.name' => Db}};
+
+handler_info('POST', [Db, <<"_changes">>], _) ->
+    {'db.changes.read', #{'db.name' => Db}};
+
+handler_info('POST', [Db, <<"_compact">>], _) ->
+    {'db.compact.execute', #{'db.name' => Db}};
```

Comment:
There is view compaction too... Not sure if we wanted capture it as well.

On the other hand, compaction is not a thing in 4.x. Do we want to exclude all the old API endpoints?

Reply:
I did all of these by combing through the existing chttpd_handler plugins. I didn't put in any effort thinking about what may or may not exist in the future.